### PR TITLE
ALSA: hda: MTL: add HD Audio PCI ID and HDMI codec vendor ID

### PR DESCRIPTION
**Pull request by yongzhi1**
Add HD Audio PCI ID for Intel Meteorlake platform.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>
Signed-off-by: Yong Zhi <yong.zhi@intel.com>

### CODE REVIEW

File: sound/pci/hda/hda_intel.c [hunk 1]
```diff
@@ -2509,6 +2509,9 @@ static const struct pci_device_id azx_ids[] = {
 	/* Alderlake-N */
 	{ PCI_DEVICE(0x8086, 0x54c8),
 	  .driver_data = AZX_DRIVER_SKL | AZX_DCAPS_INTEL_SKYLAKE},
+	/* Meteorlake-P */
+	{ PCI_DEVICE(0x8086, 0x7E28),
+	  .driver_data = AZX_DRIVER_SKL | AZX_DCAPS_INTEL_SKYLAKE},
```
Comment (kv2019i) [inline sound/pci/hda/hda_intel.c]: @yongzhi1 Could you actually move this as the last DCAPS_INTEL_SKYLAKE platform, i.e. between Raptor Lake and Broxton-P? The entries are not well ordered, but to retain some logic, adding new platforms at the end of the list makes sense.

File: sound/pci/hda/patch_hdmi.c [hunk 2]
```diff
@@ -4551,6 +4551,7 @@ HDA_CODEC_ENTRY(0x80862819, "DG2 HDMI",	patch_i915_adlp_hdmi),
 HDA_CODEC_ENTRY(0x8086281a, "Jasperlake HDMI",	patch_i915_icl_hdmi),
 HDA_CODEC_ENTRY(0x8086281b, "Elkhartlake HDMI",	patch_i915_icl_hdmi),
 HDA_CODEC_ENTRY(0x8086281c, "Alderlake-P HDMI", patch_i915_adlp_hdmi),
+HDA_CODEC_ENTRY(0x8086281d, "Meteorlake HDMI",	patch_i915_tgl_hdmi),
```
Comment (kv2019i) [inline sound/pci/hda/patch_hdmi.c]: Please use "patch_i915_adlp_hdmi" now that it is available.

File: sound/pci/hda/hda_intel.c (hunk 1 above)
Comment (yongzhi1) [reply sound/pci/hda/hda_intel.c]: Ack, will update

File: sound/pci/hda/patch_hdmi.c (hunk 2 above)
Comment (yongzhi1) [reply sound/pci/hda/patch_hdmi.c]: Ack

**Comment by yongzhi1** [comment]
V2 update:
1. Re-order PCI_DEVICE entry for MTL
2. Use patch_i915_adlp_hdmi() for .driver_data of MTL HDA_CODEC_ENTRY().

**Review by plbossart — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Pull request merged**